api/src/main/java/au/org/aodn/nrmn/restapi/service/WormsService.java

```diff
@@ -21,7 +21,7 @@
 @Service
 public class WormsService {
 
-    static private Pattern REMOVE_TRAILING_JUNK_PATTERN = Pattern.compile("( (sp\\.|spp\\.))?( (\\(.*\\)|\\[.*\\]))?$",
+    static private Pattern REMOVE_TRAILING_JUNK_PATTERN = Pattern.compile("(( (sp\\.|spp\\.).+)|( \\(.+)|( \\[.+))",
```

Comment:
Can you add a test case for this reg expression as it is hard to verify later what you want to filter out what not.

Reply:
@utas-raymondng here is it: https://github.com/aodn/nrmn-application/blob/master/api/src/test/java/au/org/aodn/nrmn/restapi/service/WormsServiceTest.java